# Assignment Day_20

__Question 1:__

__Anova__

Three archers – Pat, Jack, and Alex are participating in an archery contest. They are shooting at

targets with 10 evenly spaced concentric rings. The rings have score values from 1 through 10

assigned to them, with 10 being the highest. Each participant shoots 6 arrow, scoring the

following points:

Pat – 5, 4, 4, 3, 9, 4

Jack – 4, 8, 7, 5, 1, 5

Alex – 9, 9, 8, 10, 4, 10

Based on the above results we would like to know who the best archer is. In other words our null

hypothesis is that the means of all populations are equal.

In [1]:
import pandas as pd

In [2]:
Pat = [5, 4, 4, 3, 9, 4]

Jack = [4, 8, 7, 5, 1, 5]

Alex = [9, 9, 8, 10, 4, 10]

In [3]:
all_scores = Pat + Jack + Alex
player_name = (['Pat'] * len(Pat)) +  (['Jack'] * len(Jack)) +  (['Alex'] * len(Alex))

In [4]:
data = pd.DataFrame({'player_name': player_name, 'score': all_scores})

In [5]:
data


player_name  score
0          Pat      5
1          Pat      4
2          Pat      4
3          Pat      3
4          Pat      9
5          Pat      4
6         Jack      4
7         Jack      8
8         Jack      7
9         Jack      5
10        Jack      1
11        Jack      5
12        Alex      9
13        Alex      9
14        Alex      8
15        Alex     10
16        Alex      4
17        Alex     10

In [6]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [7]:
lm = ols('score ~ player_name',data=data).fit()
table = sm.stats.anova_lm(lm)
print(table)

               df     sum_sq    mean_sq         F    PR(>F)
player_name   2.0  46.777778  23.388889  4.488273  0.029668
Residual     15.0  78.166667   5.211111       NaN       NaN


__As 0.02(P value) less than 0.04(F Value) we reject the null hypothesis and we conclude that at least one of the means is different from at least one other population mean (i.e. not all archers perform equally)__

__Tukey’s range test__

In [8]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
from statsmodels.stats.multicomp import MultiComparison

In [9]:
mc = MultiComparison(data['score'], data['player_name'])
result = mc.tukeyhsd()
 
print(result)
print(mc.groupsunique)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj   lower  upper  reject
---------------------------------------------------
  Alex   Jack  -3.3333 0.0568 -6.7554 0.0887  False
  Alex    Pat     -3.5 0.0447  -6.922 -0.078   True
  Jack    Pat  -0.1667    0.9 -3.5887 3.2554  False
---------------------------------------------------
['Alex' 'Jack' 'Pat']


The results above reveal that Alex  significantly differs from the other two archers.

The third column tells us that there is significant evidence to reject the null hypothesis for the groups Alex-Jack and 

Alex-Pat.

The test also shows the difference between the group means (the meandiff column).

    [mean{Jack} - mean{Alex} = -3.3333]

    [mean{Pat} - mean{Alex} = - 3.5]

This leads to the conclusion that Alex is the best archer in the group.